File: sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int p1=0,curr=0,p2=nums.size()-1;
        while(curr<=p2){
            if(nums[curr]==0){
                swap(nums[curr],nums[p1]);
                curr++;
                p1++;
            }
            else if(nums[curr]==2){
                swap(nums[curr],nums[p2]);
                p2--;
            }
            else
                curr++;
        }
    }
};

File: sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int count_0=0,count_1=0,n=nums.length;
        for(int i=0;i<n;i++){
            if(nums[i]==0)
                count_0++;
            else if(nums[i]==1)
                count_1++;
        }
        for(int i=0;i<count_0;i++){
            nums[i]=0;
        }
        for(int i=count_0;i<count_1+count_0;i++){
            nums[i]=1;
        }
        for(int i=count_1+count_0;i<n;i++){
            nums[i]=2;
        }
    }
}